File: scripts/algorithms/D/Duplicate Zeros/Duplicate Zeros.py
# Runtime: 107 ms (Top 60.66%) | Memory: 14.9 MB (Top 70.68%)
class Solution:
    def duplicateZeros(self, arr: List[int]) -> None:
        """
        Do not return anything, modify arr in-place instead.
        """
        l = len(arr)
        i,c=0,0
        while i<l:
            if arr[i]==0:
                arr.insert(i+1,0)
                i+=1
                arr.pop()
            i+=1

File: scripts/algorithms/D/Duplicate Zeros/Duplicate Zeros.cpp
// Runtime: 87 ms (Top 26.83%) | Memory: 9.5 MB (Top 93.67%)
class Solution {
public:
    void duplicateZeros(vector<int>& arr) {
        for(int i=0;i<arr.size();i++)
        {
          if(arr[i]==0)
          {
              arr.pop_back();
              arr.insert(arr.begin()+i,0);
              i++;
          }
            }

    }
};